PantheonCMD/pcbuild.py

```diff
@@ -82,28 +98,52 @@ def prepare_build_directory():
     os.makedirs('build/images')
 
 
-def coalesce_document(main_file, attributes=None):
+def resolve_attribute_tree(content, attributes):
+    """Attempt to recursively resolve AsciiDoc attributes into a string."""
+
+    continue_processing = True
+
+    while continue_processing:
+        if re.match(r'.*\{\S+\}.*', content):
+            for attribute in re.search(r'\{(.*?)\}', content).groups():
+                if attribute in attributes.keys():
+                    content = content.replace('{' + attribute + '}',attributes[attribute])
+                else:
+                    continue_processing = False
+        else:
+            continue_processing = False
+    return content
+
+
+def coalesce_document(main_file, attributes=None, depth=1, top_level=True):
     """Combines the content from includes into a single, contiguous source file."""
     attributes = attributes or {}
     lines = []
 
     # Open the file, iterate over lines
     if os.path.exists(main_file):
         with open(main_file) as input_file:
+            # Iterate over content
             for line in input_file:
                 # Process comments
                 if line.startswith('//'):
                     continue
+                # Process section depth
+                if line.strip().startswith('= '):
+                    lines.append(('=' * depth) + ' ' + line.split('= ')[1])
                 # Process includes - recusrive
-                if line.startswith("include::"):
+                elif line.startswith("include::"):
+                    include_depth = 1
                     include_file = line.replace("include::", "").split("[")[0]
-                    # Replace attributes in includes, if already detected
+                    include_options = line.split("[")[1].split("]")[0]
+                    if include_options.__contains__("="):
```

Comment:
Same as one of the previous comments about authors uses level 2 headings

Reply:
Great catch, @Levi-Leah - I've address this one above, so I'll resolve this comment and follow on with the conversation there.